adding more information about Interfaces in Java (#32539)
freeCodeCamp/freeCodeCamp: guide/english/java/data-abstraction/index.md

In Java, abstraction is achieved using Abstract classes and interfaces.

Interfaces were introduced in Java to enhance Java’s single-inheritance model, multiple inheritance created too many problems for programmers and compiler writers. 

> All methods in interface are public, we may not write it every time, like all of them are abstract, but we never write the key word abstract.

Reviewer: ```suggestion
All methods in an Interface are implicitly `public` so we need not write the `public` keyword each time. 
```

Java 8 introduced default methods, which can contain full implementation, so "all of them are abstract" doesn't really hold true anymore.

> The interface should have at least one method but not too many.

Reviewer: Is there a reference that you can link to which supports this statement?

Advantages using interfaces:

> It decouples your code.

Reviewer: Please use list formatting to list out the advantages. Please review the [style guide for guide articles](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/style-guide-for-guide-articles.md) if you need help with formatting.
